app/controllers/authenticate_controller.rb

```diff
@@ -136,39 +140,68 @@ def authenticate_jwt
     handle_authentication_error(e)
   end
 
+  def logout_oidc
+    params[:authenticator] = "authn-oidc"
+    decode_and_merge_request_body
+    params.permit!
+
+    logout_uri = Authentication::Handler::LogoutHandler.new(
+      authenticator_type: params[:authenticator]
+    ).call(
+      parameters: params.to_hash.symbolize_keys,
+      request_ip: request.ip
+    ) do |authenticator|
+      Authentication::AuthnOidc::V2::Strategies::Logout.new(
+        authenticator: authenticator
+      ).callback(params.to_hash.symbolize_keys)
+    end
+
+    render(json: { oidc_logout_uri: logout_uri })
+  end
+
   def authenticate_oidc
     params[:authenticator] = "authn-oidc"
     decode_and_merge_request_body
-    
-    if params[:refresh_token] || params[:code]
-      params.permit!
+    params.permit!
+
+    if params[:code] || params[:refresh_token]
+      namespace = 'Authentication::AuthnOidc::V2::Strategies'
+      strategy = 'AuthzCode' if params[:code]
```

Comment:
I'd actually recommend pushing the logic for choosing a strategy up into the `Authentication::AuthnOidc::V2::Strategy` class.  This will allow you to keep details of a particular strategy out of the controller (which ultimately should be super generic).  Something like:

```ruby
module Authentication
  module AuthnOidc
    module V2
      class Strategy
        def callback(args)
          if args.key?(:refresh_token)
            # run refresh token workflow
          elsif(args.key?(:code)
            # run code validation workflow
          end
          ...
        end
      end
    end
  end
end
```